Homework3/Novikov_homework3.cpp

```diff
@@ -0,0 +1,262 @@
+#include <iostream>
+#include <random>
+
+bool checkString(char *string, int maxSize) {
+  if (string != nullptr) {
+    while (true) {
+      if (!std::cin.getline(string, maxSize)) {
+        std::cout << "Error in string range, try again\n";
+        std::cin.clear();
+        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+      } else if (*string != 0)
+        return true;
+      std::cout << "Empty string try again\n";
+    }
+  }
+  return false;
+}
+
+void outputArray(int *begin, int *end) {
+  if (begin == nullptr || end == nullptr)
+    return;
+  while (begin != end)
+    std::cout << *begin++ << " ";
+}
+
+void TaskOneReverse() {
+  // clear cin buffer after last input
+  std::cin.clear();
+  std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+  constexpr int kMaxStringSize{255};
+  char string[kMaxStringSize]{};
+  std::cout << "Enter string less than " << kMaxStringSize << " symbol\n";
+
+  if (!checkString(string, kMaxStringSize)) {
+    std::cout << "Error in string input\n";
+    return;
+  }
+
+  const int cycleDuration = strlen(string) / 2;
+  int endOfString = strlen(string) - 1;
+
+  for (size_t i = 0; i < cycleDuration; i++, endOfString--) {
+    char buf = string[i];
+    string[i] = string[endOfString];
+    string[endOfString] = buf;
+  }
+  std::cout << "Result string: " << string;
+}
+
+void TaskTwoInsertion() {
+  constexpr int arrayMaxSize{5};
+  int MyArray[arrayMaxSize]{};
+
+  std::cout << "Enter array elements\n";
+  for (size_t i = 0; i < arrayMaxSize; i++) {
+    std::cin >> MyArray[i];
+    for (int j = i; j > 0; j--) {
+      if (MyArray[j] > MyArray[j - 1]) {
+        break;
+      }
+      int buf = MyArray[j];
+      MyArray[j] = MyArray[j - 1];
+      MyArray[j - 1] = buf;
+    }
+  }
+
+  int *MyArrayBegin = std::begin(MyArray);
+  int *MyArrayEnd = std::end(MyArray);
+  std::cout << "Array: [ ";
+  outputArray(MyArrayBegin, MyArrayEnd);
+  std::cout << "]\n";
+}
+
+void deleteElement(int *array, int number, int &arraySize) {
+  for (size_t i = number; i < arraySize; i++)
+    array[i] = array[i + 1];
+  arraySize--;
+}
+
+void findElement(int *array, int &arraySize) {
+  int finder{};
+  std::cout << "Enter element to find \n";
+  std::cin >> finder;
+  if (arraySize > 0) {
+    for (size_t i = 0; i < arraySize; i++) {
+      if (array[i] == finder) {
+        deleteElement(array, i, arraySize);
+        i--;
+      }
+    }
+  } else
+    std::cout << "No elements left\n";
+}
+
+void TaskThreeDeleteElement() {
+  constexpr int maxArraySize = 10;
+  int ArraySize = maxArraySize;
+  int MyArray[maxArraySize];
```

Comment:
```suggestion
  int MyArray[maxArraySize]{};
```
это для уверености инициализации элементов массива